core/store/src/migrations.rs

```diff
@@ -414,7 +430,65 @@ pub fn migrate_41_to_42(store: &Store) -> anyhow::Result<()> {
         if epoch_first_block != sync_hash {
             tracing::warn!(key = %epoch_first_block, %sync_hash, "sync_hash field of legacy StateSyncInfo not equal to the key. Something is wrong with this node's catchup info");
         }
-        let new_info = StateSyncInfo::V0(StateSyncInfoV0 { sync_hash, shards });
+        let new_info = LegacyStateSyncInfo::V0(LegacyStateSyncInfoV0 { sync_hash, shards });
+        update
+            .set_ser(DBCol::StateDlInfos, &key, &new_info)
+            .context("failed writing to StateDlInfos")?;
+    }
+    update.commit()?;
+    Ok(())
+}
+
+/// Migrates the database from version 42 to 43.
+///
+/// This rewrites the contents of the StateDlInfos column
+pub fn migrate_42_to_43(store: &Store) -> anyhow::Result<()> {
+    let mut update = store.store_update();
+
+    for row in store.iter_ser::<LegacyStateSyncInfo>(DBCol::StateDlInfos) {
+        let (key, info) =
+            row.context("failed deserializing legacy StateSyncInfo in StateDlInfos")?;
```

Comment:
Nice, we should use context more. Is it available on all errors or just anyhow?

Reply:
Yeah it's just an anyhow thing. I really wonder if we should use these more throughout the code. There are places where we have a big enum like `near_chain::Error` that we take pains to initialize to some particular variant, but we just use `?` to pass errors up the stack and then log it at the top. Seems like we should just be using anyhow in those cases in my opinion. Because then for example that one recurrent bug that sometimes happens to people where all you see is something like `called Result::unwrap() on an Err value: Chain(DBNotFoundErr("epoch block .."))` would actually have more info saying what rly happened